### PR TITLE
figure option dialog does not properly handle units

### PR DESCRIPTION
**Pull request by jrevans**
The figure option dialog resets the axes and all unit information.  This saves the axes converter information so that it can be restored when the dialog values are applied to the figure.

This addresses an issue in #4897.

### CODE REVIEW

File: lib/matplotlib/backends/qt_editor/figureoptions.py [hunk 1]
```diff
@@ -122,6 +128,14 @@ def apply_callback(data):
         axes.set_ylim(ymin, ymax)
         axes.set_ylabel(ylabel)
 
+        # Restore the unit data
+        axes.xaxis.converter = xconverter
+        axes.yaxis.converter = yconverter
+        axes.xaxis.set_units( xunits )
+        axes.yaxis.set_units( yunits )
```
Comment (WeatherGod) [inline lib/matplotlib/backends/qt_editor/figureoptions.py]: PEP8: unnecessary spaces

**Comment by WeatherGod** [comment]
Travis is only complaining about the PEP8 errors I noted. The change makes sense.

At some point, I should update the editor to be mplot3d-friendly...

File: lib/matplotlib/backends/qt_editor/figureoptions.py [hunk 2]
```diff
@@ -101,7 +107,7 @@ def figure_edit(axes, parent=None):
         has_curve = bool(curves)
 
     datalist = [(general, "Axes", "")]
-    if has_curve:
+    if has_curve and curves:
```
Comment (tacaswell) [inline lib/matplotlib/backends/qt_editor/figureoptions.py]: What is the `and curves` doing?

**Comment by tacaswell** [comment]
What mechanism in that call back is re-setting the converters/units?   That seems like the actually and this is just treating the symptoms.

**Comment by jrevans** [comment]
The spacing is now cleaned up as per your style rules.  As for the root of the problem, when the apply callback happens it takes the values for the limits specified in the dialog (which do not have any unit information) and uses those to 'set_xlim' and 'set_ylim', which will set the units on the respective axis based on the values passed in.  Since the limit values have become unitless in the GUI they stay unitless during the 'set_*' call and the axes forget what the units should be.  This will preserve that unit information and make sure it stay in place after an apply is called.

As for the 'and curves', If you have a plot with a line, whose label is set to '_nolegend_', then the logic fails as 'curves' is an empty list.  I opted for the simplest fix of making sure that 'curves' was not empty after the preprocessing step (in the event the logic of the preprocessing step changes in the future).

**Comment by tacaswell** [comment]
I think the issue with an empty curve list was fixed in https://github.com/matplotlib/matplotlib/pull/4411 (originally reported as https://github.com/matplotlib/matplotlib/issues/4323).  Can you please make sure that change is actually needed?

**Pull request merged**